osquery/carver/carver_utils.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#ifdef WIN32
+#define _WIN32_DCOM
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#endif
+
+// This define is required for Windows static linking of libarchive
+#define LIBARCHIVE_STATIC
+#include <archive.h>
+#include <archive_entry.h>
+#include <zstd.h>
+
+#include <osquery/system.h>
+
+#include "osquery/carver/carver.h"
+#include "osquery/filesystem/fileops.h"
+
+namespace osquery {
+
+Status compress(const boost::filesystem::path& in,
+                const boost::filesystem::path& out) {
+  PlatformFile archFile(in.string(), PF_OPEN_EXISTING | PF_READ);
```

Comment:
nit: would you mind using naming schemes that don't have `arch` in them? In the off chance folks are using these APIs without a `.tar` file it might make a little more sense.